workers/default/xml/tasks.py

```diff
@@ -56,29 +28,47 @@ class GenerateXMLTask(ObjectTask):
     name = "generate_xml"
 
     def process_object(self, obj):
-        articles_meta = []
-        for part in obj.get_parts():
-            articles_meta.append({
-                **part.metadata.to_dict(),
-                'filepath': os.path.join(part.get_representation_dir(
-                                Object.INITIAL_REPRESENTATION),
-                            'merged.pdf')  # TODO filename may be different?
-            })
+        xml_type = self.get_param('xml_type')
+
+        if xml_type == 'ojs':
+            xml_template_string = _read_file('resources/ojs_template.xml')
+            target_filename = 'ojs_import.xml'
+            dtd_file = 'resources/ojs_import.dtd'
+            schema_file = None
+        elif xml_type == 'marc':
+            xml_template_string = _read_file('resources/marc_template.xml')
+            target_filename = 'marc.xml'
+            dtd_file = None
+            schema_file = 'resources/MARC21slim.xsd'
+        else:
+            raise ValueError(f"Unknown XML type: {{xml_type}}")
 
         data = {
-            'data': {
-                **obj.metadata.to_dict(),
-                **self.get_param('ojs_metadata')
-            },
-            'articles': articles_meta}
+            'metadata': obj.metadata.to_dict()
+            }
+
+        # parts mean that this is an issue, and not an article
+        if any(obj.get_parts()):
+            articles_meta = []
+            # add filepath to article data, which is needed in the XML
+            for part in obj.get_parts():
+                articles_meta.append({
+                    **part.metadata.to_dict(),
+                    'filepath': os.path.join(part.get_representation_dir(
+                                             Object.INITIAL_REPRESENTATION),
+                                             'merged.pdf')  # TODO filename may be different?
```

Comment:
Auch hier: Das ist sehr auf den OJS-Fall zugeschnitten und nicht wirklich generisch.

Was mich auch gerade frage: Wo wird denn überhaupt die ojs-XML für das Issue generiert? Wenn ich das richtig sehe, ist in der ingest_journal.yml jetzt nur die Generierung der ojs_import.xml für die parts definiert, oder?

Reply:
`merged.pdf` sollte ins Template und der `object_path` als Parameter ans Template übergeben werden.